proof_ipa.go

```diff
@@ -79,6 +79,7 @@ func GetCommitmentsForMultiproof(root VerkleNode, keys [][]byte) (*ProofElements
 	return root.GetProofItems(keylist(keys))
 }
 
+// QUESTION(jsign): do we need this `keys` and `keyvals` separation? Shouldn't keys == keys(keyvals)?
 func MakeVerkleMultiProof(root VerkleNode, keys [][]byte, keyvals map[string][]byte) (*Proof, []*Point, []byte, []*Fr, error) {
```

Comment:
@gballet, why we'd want a `keyvals` map that contains key-values that aren't in `keys`?
The method seems to ignore keys in `keyvals` that aren't in `keys`.

Also, in L103 there's an assumption that `keyvals[string(k)]` will return a value; but technically it could return a default value if the key isn't present in the map. Feels like an obscure assumption too.